diff --git a/src/pffft-avx.h b/src/pffft-avx.h
index ace19b5..8dd9b4f 100644
--- a/src/pffft-avx.h
+++ b/src/pffft-avx.h
@@ -26,10 +26,10 @@ typedef __m256d v4sf;
   out1 = _mm256_unpacklo_pd(t1, t2); \
   out2 = _mm256_unpackhi_pd(t1, t2);}
 #define VTRANSPOSE4(x0,x1,x2,x3) {v4sf \
-  t0 = _mm256_shuffle_pd(x0,x1, 0x0), \
-  t2 = _mm256_shuffle_pd(x0,x1, 0xf), \
-  t1 = _mm256_shuffle_pd(x2,x3, 0x0), \
-  t3 = _mm256_shuffle_pd(x2,x3, 0xf); \
+  t0 = _mm256_shuffle_pd(x0,x1, 0x00), \
+  t2 = _mm256_shuffle_pd(x0,x1, 0x0F), \
+  t1 = _mm256_shuffle_pd(x2,x3, 0x00), \
+  t3 = _mm256_shuffle_pd(x2,x3, 0x0F); \
   x0 = _mm256_permute2f128_pd(t0,t1, 0x20); \
   x1 = _mm256_permute2f128_pd(t2,t3, 0x20); \
   x2 = _mm256_permute2f128_pd(t0,t1, 0x31); \
diff --git a/src/soxr.c b/src/soxr.c
index c2861ac..8afbbfd 100644
--- a/src/soxr.c
+++ b/src/soxr.c
@@ -815,7 +815,7 @@ soxr_error_t soxr_oneshot(
     soxr_quality_spec_t const * q_spec,
     soxr_runtime_spec_t const * runtime_spec)
 {
-  soxr_t resampler;
+  soxr_t resampler = NULL;
   soxr_error_t error = q_spec? q_spec->e : 0;
   if (!error) {
     soxr_quality_spec_t q_spec1;
